tests/sentry/api/test_event_search.py

```diff
@@ -686,6 +686,61 @@ def test_basic_fallthrough(self):
             assert parse_search_query(query) == [expected]
 
 
+class ParseBooleanSearchQueryTest(TestCase):
+    def setUp(self):
+        super(ParseBooleanSearchQueryTest, self).setUp()
+        self.left_term = SearchFilter(
+            key=SearchKey(name='user.email'),
+            operator="=",
+            value=SearchValue(raw_value='foo@example.com'),
+        )
+        self.right_term = SearchFilter(
+            key=SearchKey(name='user.email'),
+            operator="=",
+            value=SearchValue(raw_value='bar@example.com'),
+        )
+        self.term3 = SearchFilter(
+            key=SearchKey(name='user.email'),
+            operator="=",
+            value=SearchValue(raw_value='foobar@example.com'),
+        )
+
+    def test_simple(self):
+        assert parse_search_query(
+            'user.email:foo@example.com OR user.email:bar@example.com'
+        ) == [SearchBoolean(left_term=self.left_term, operator="OR", right_term=self.right_term)]
+
+        assert parse_search_query(
+            'user.email:foo@example.com AND user.email:bar@example.com'
+        ) == [SearchBoolean(left_term=self.left_term, operator="AND", right_term=self.right_term)]
+
+    def test_single_term(self):
+        assert parse_search_query('user.email:foo@example.com') == [self.left_term]
+
+    def test_multiple_statements(self):
+        assert parse_search_query(
+            'user.email:foo@example.com OR user.email:bar@example.com OR user.email:foobar@example.com'
+        ) == [SearchBoolean(
+            left_term=SearchBoolean(
+                left_term=self.left_term,
+                operator="OR",
+                right_term=self.right_term),
+            operator="OR",
+            right_term=self.term3
+        )]
+
+        assert parse_search_query(
+            'user.email:foo@example.com AND user.email:bar@example.com AND user.email:foobar@example.com'
```

Comment:
Should there be a test covering a mixture of OR & AND operators used together?

Reply:
No this PR only addresses OR and AND separately. OR and AND together will require grouping to be what you probably actually mean. I'm going to follow up with another pr to implement grouping.

Reply:
You know what nevermind. I can add tests without grouping. They probably are not going to be examples of what the user was intending... but it'll be good to add the tests anyway.